networking_oneview/ml2/drivers/oneview/neutron_oneview_client.py

```diff
@@ -183,35 +181,61 @@ def _create_network_on_oneview(self, name, network_type, seg_id):
         }
         return self.oneview_client.ethernet_networks.create(options)
 
+    def _add_to_ligs(self, network_type, physical_network, oneview_network):
+        try:
+            lig_list = self._get_lig_list(physical_network, network_type)
+            uplinksets_list = self._get_uplinksets_from_lig(
+                network_type, lig_list)
+            self._add_network_to_logical_interconnect_group(
+                lig_list, oneview_network.get('uri'))
+            self._add_network_to_uplink_sets(
+                uplinksets_list, oneview_network.get('uri'))
+            return lig_list
+        except exceptions.NetworkCreationException:
+            return None
+
     def _add_network_to_logical_interconnect_group(
             self, uplinkset_mappings, network_uri):
         for lig_id, uplinkset_name in zip(
                 uplinkset_mappings[0::2], uplinkset_mappings[1::2]):
             logical_interconnect_group = (
                 common.get_logical_interconnect_group_by_id(
-                    self.oneview_client, lig_id
-                )
-            )
+                    self.oneview_client, lig_id))
             lig_uplinksets = logical_interconnect_group.get('uplinkSets')
+            lig_uri = logical_interconnect_group.get('uri')
             uplinkset = common.get_uplinkset_by_name_from_list(
-                lig_uplinksets, uplinkset_name
-            )
+                lig_uplinksets, uplinkset_name)
             if network_uri not in uplinkset['networkUris']:
-                uplinkset['networkUris'].append(network_uri)
-                self.oneview_client.logical_interconnect_groups.update(
-                    logical_interconnect_group
-                )
-
-    def _add_network_to_logical_interconnects(
-            self, uplinkset_list, network_uri):
+                try:
+                    uplinkset['networkUris'].append(network_uri)
```

Comment:
Do we need to update the lig's uplinkset list manually? In the method _add_network_to_uplink_sets, doesn't it update the uplinkset and therefore the lig's uplinkset list?

Reply:
This will get only one uplinkset.

Reply:
@stenioaraujo is this what we talked about yesterday (the uplinkset from LIG is a different object than the uplinkset from LI)?